### DISCUSSION TITLE
Implement Ollama into dnSpy

### DISCUSSION THREAD

**Discussion by hasokeric** [dnSpyEx/dnSpy#357] [Ideas]
What are your thoughts on adding some Ollama Chat sidebar to dnSpy so you can easily chat about the file you are debugging, decompiling. I was going to suggest ChatGPT but it might not want to help with obfuscated (decompiled) code.

**Comment by ElektroKill** [comment]
Hello, AI will probably not be implemented natively into dnSpy. Using online services means having some sort of privacy policy and would require an API token. Local AI will probably not work on all machines dnSpy runs on so it's also not a possibility. However, it is possible to code an extension that does just this using the extension API.
